crates/uv-resolver/src/lock.rs

```diff
@@ -174,8 +174,8 @@ impl TryFrom<LockWire> for Lock {
                 }
             }
             for wheel in &dist.wheels {
-                if requires_hash != wheel.hash.is_some() {
-                    return Err(LockError::hash(dist.id.clone(), "wheel", requires_hash));
+                if wheel.hash.is_none() {
+                    return Err(LockError::hash(dist.id.clone(), "wheel", true));
```

Comment:
So I basically agree with the end effect here, but I don't think this is actually internally consistent. The reason why I made it possible for hash checking in wheels to be optional is because this is what the data model currently supports:

https://github.com/astral-sh/uv/blob/eab2b832a65f7957ce073046f39131592fb566e2/crates/uv-resolver/src/lock.rs#L739-L745

The problem here, as far as I can tell, is that not all "built" distributions have hashes attached to them. So when they make it to the lock file data structure, there's no hash to include.

What this means is that this checking will reject lock files that can be generated right now.

With that said, I do agree that this change is, I think, what we want our _end state_ to be. I don't think I feel strongly about whether we do the stricter checking now, but I did kind of feel like the checking should be consistent with our serialization. If not, then perhaps a comment here explaining the inconsistency would be helpful.